plaidml/edsl/__init__.py

```diff
@@ -600,6 +600,40 @@ def TensorIndexes(count):
     return [TensorIndex() for i in range(count)]
 
 
+def Placeholder(*args, **kwargs):
+    """Creates a placeholder tensor, specified either by a combination of data
+       type and dimensions or by passing in a LogicalShape.
+
+        Args:
+            dtype (DType): Specifies the data type of the ``Placeholder``.
+            dims (list): Specifies the dimensions of the ``Placeholder``.
+            shape (LogicalShape): Specifies the shape of the ``Placeholder``.
```

Comment:
This isn't quite right, because it's not actually the case that users will do:

```
Placeholder(dtype, dims, shape)
```

There are 2 choices:

```
Placeholder(dtype, dims)
Placeholder(shape)
```

To resolve this, let's do this instead:

```
def Placeholder(dtype_or_shape, dims=[], name=''):
```

This then removes much of the bolierplate to determine how the `Placeholder` was called. The only thing to determine is whether `dtype_or_shape` is indeed a `plaidml.DType` or a `plaidml.LogicalShape`